### DISCUSSION TITLE
Running user created query

### DISCUSSION THREAD

**Discussion by felipesere** [nvim-treesitter/nvim-treesitter#7442] [Q&A]
This might be obvious but I couldn't tell how to run a query from lua that the user has written.

My goal is to let the user write something like `foo.bar[1].baz`, turn that into a query and if its found call `goto_node(n)` to place the cursor there.
This is particularly useful for large YAML, XML, and JSON files.

**Comment by clason** [comment]
`:h vim.treesitter.query.set()`

**Comment by felipesere** [comment]
Oh that was quick! I'll have a play with it tonight 🙇
